src/Mvc/Mvc.Analyzers/src/Startup/MiddlewareAnalyzer.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.AspNetCore.Analyzers
+{
+    internal class MiddlewareAnalyzer
```

Comment:
It's a common pattern writing stateful analyzers to call things `FooAnalyzer` even though they aren't registered as top-level analyzers.